### PR TITLE
feat(cli): call watcher close at process exit

### PR DESCRIPTION
**Pull request by underfin**
### Description

Inspired rollup implementation, ref https://github.com/rollup/rollup/blob/master/cli/run/watch-cli.ts#L31

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
Can we add a test to ensure that if there are any errors, the watcher can be closed with a zero exit code?

File: packages/rolldown/package.json [hunk 1]
```diff
@@ -101,7 +101,8 @@
     "dtsHeader": "type MaybePromise<T> = T | Promise<T>\ntype Nullable<T> = T | null | undefined\ntype VoidNullable<T = void> = T | null | undefined | void\nexport type BindingStringOrRegex = string | RegExp\n\n"
   },
   "dependencies": {
-    "zod": "^3.23.8"
+    "zod": "^3.23.8",
+    "signal-exit": "4.1.0"
```
Comment (hyf0) [inline packages/rolldown/package.json]: Better put it in `devDependencies`, since we will inline this package during bundling.

Comment (underfin) [reply packages/rolldown/package.json]: Done

**Comment by underfin** [comment]
> Can we add a test to ensure that if there are any errors, the watcher can be closed with a zero exit code?

I tested it, because the `watcher.close` is a async,`onExit` maybe could't wait `watcher.close` finish, it is unstable. It is difficult to add a stable test for it, so let us skip it at now.

**Comment by underfin** [comment]
I get some info after research, it is difficult to ensure async task is finished at process exit. But here has a helper info us to out. 

If you ^C a process, the kernel sends a SIGINT to the process group; that usually terminates both the parent and its children, grandchildren, and so on.

So we only need to care about is using watch cli bin at child process, Like test watch-cli using child process, it is unstable at process exit. So here i suggest to skip it at now.

**Review by IWANABETHATGUY — APPROVED** [review]

**Comment by underfin** [comment]
tracking test at https://github.com/rolldown/rolldown/issues/2584
